### ISSUE TITLE
Issue with Invoke-DbaDataMasking

### ISSUE THREAD

**Issue by rvftskill** [sqlcollaborative/dbatools#5248]
### Before submitting a bug report:

- [X] Ensure you are able to reproduce it on the latest released version (_we release often_)
- [X] Verified this bug is not already reported in an issue
- [X] Verified errors are not related to permissions
- [x] Can reproduce in a clean PowerShell session (_clean = `powershell -NoProfile`_)

## Steps to Reproduce

```
Invoke-DbaDbDataMasking -SqlInstance SERVER\MSSQLSERVER -Database test -FilePath "C:\scripts\anon.json"
```

## Expected Behavior

The masking must be started.

## Actual Behavior

I have this error 
```
You cannot call a method on a null-valued expression.
At C:\Program Files\WindowsPowerShell\Modules\dbatools\0.9.789\allcommands.ps1:58468 char:93
+ ...  = $faker.$($columnMaskInfo.MaskingType).$($columnMaskInfo.SubType)()
+                                                ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
```
## Environmental data
```
Powershell 

Name                           Value                                                                                                                                                                                                                 
PSVersion                      5.1.14393.2125                                                                                                                                                 
PSEdition                      Desktop                                                                                                                                                        
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                        
BuildVersion                   10.0.14393.2125                                                                                                                                                
CLRVersion                     4.0.30319.42000                                                                                                                                                
WSManStackVersion              3.0                                                                                                                                                            
PSRemotingProtocolVersion      2.3                                                                                                                                                            
SerializationVersion           1.1.0.1

```
```
SQL Version

 Microsoft SQL Server 2017 (RTM) - 14.0.1000.169 (X64)   Aug 22 2017 17:04:49   Copyright (C) 2017 Microsoft Corporation  Enterprise Edition: Core-based Licensing (64-bit) on Windows Server 2016 Datacenter 10.0 <X64> (Build 14393: ) (Hypervisor) 
```

## Complementary information

I have tested many configurations :

* Database with a modified json configuration file -> Not working
* Database with a non modified json configuration file -> Not working
* The Database "AdventureWork2014" with the default json config file -> Worked

**Comment by rvftskill** [comment]
Okay I figure out the issue : On the generated JSON file, there is the option "Has Unique Index : true". I deleted it and it's worked.

But the removal of this option can cause some issues after no ?

**Comment by potatoqualitee** [comment]
thank you for the report!

**Comment by SQLGB** [comment]
Same behavior in 0.9.803.

**Comment by sanderstad** [comment]
@SQLGB Is there a way you can check the changes made by using the development version of dbatools?
And especially using the branch "fix-masking"

**Comment by SQLGB** [comment]
@sanderstad I'll attempt to do that and share results later.

**Comment by zandhaai** [comment]
For me the above command does not work on all databases I tried. I tried it on 1 of my own database, AdventureWorks2014, AdventureWorks2017. I tried it with the Install-Module version and later (by hand installation) with the dbatools-developent and fix-masking releases. 

Steps:

1. Run Powershell Command as administrator
2. Run: New-DbaDbMaskingConfig -SqlInstance localhost -Database AdventureWorks2014 -Path C:\temp\
3. Run: Invoke-DbaDbDataMasking -SqlInstance localhost -Database AdventureWorks -FilePath C:\temp\xxxxxx.AdventureWorks.tables.json

I am running SQL Server 2017 on Windows 2016, some info:

Build          : 14.0.2002
BuildLevel     : 14.0.2002
BuildTarget    : 14.0.3076
Compliant      : False
CULevel        :
CUTarget       :
KBLevel        : 4293803
MatchType      : Exact
MaxBehind      :
NameLevel      : 2017
SPLevel        : {RTM, LATEST}
SPTarget       :
SqlInstance    : xxxxxx
SupportedUntil : 10/12/2027 12:00:00 AM
Warning        :
